File: check-if-array-is-good-2784/Solution.java

/**
 * LeetCode problem # 2784, 'Check if Array is Good'
 * https://leetcode.com/problems/check-if-array-is-good/
 */

import java.util.*;

class Solution {
    public boolean isGood(int[] nums) {
        int max = 0;
        Map<Integer, Integer> map = new HashMap<>();
        for (int num : nums) {
            max = Math.max(num, max);
            map.put(num, map.getOrDefault(num, 0) + 1);
            if (map.get(num) > 1 && num != max) {
                return false;
            }
        }

        if (map.get(max) != 2) {
            return false;
        }
        return nums.length == max + 1;

    }
}

File: check-if-array-is-good-2784/solution.js
/**
 * LeetCode problem # 2784, 'Check if Array is Good'
 * https://leetcode.com/problems/check-if-array-is-good/
 */

/**
 * @param {number[]} nums
 * @return {boolean}
 */
function isGood(nums) {
  const max = Math.max(...nums);
  if (max !== nums.length - 1) {
    return false;
  }

  const map = new Map();
  for (const num of nums) {
    map.set(num, (map.get(num) ?? 0) + 1);
    if (map.get(num) > 1 && num !== max) {
      return false;
    }
  }

  if (map.get(max) !== 2) {
    return false;
  }
  return true;
}
